src/main/java/de/tum/in/www1/artemis/web/rest/ProgrammingSubmissionResource.java

```diff
@@ -186,7 +187,9 @@ public ProgrammingSubmissionResource(ProgrammingSubmissionService programmingSub
             programmingSubmissionService.notifyUserAboutSubmission(submission.get());
             return ResponseEntity.ok().build();
         }
-        if (lastGraded) {
+        if (lastGraded && submission.get().getType() != SubmissionType.INSTRUCTOR && submission.get().getType() != SubmissionType.TEST
+                && submission.get().getParticipation().getExercise().getDueDate() != null
+                && submission.get().getParticipation().getExercise().getDueDate().isBefore(ZonedDateTime.now())) {
```

Comment:
Could you add a explanation here? This is not that intuitive to understand

Reply:
Done